cli/plasmo/src/features/manifest-factory/base.ts

```diff
@@ -27,23 +27,43 @@ import {
   loadEnvConfig
 } from "~features/extension-devtools/load-env-config"
 import type { PackageJSON } from "~features/extension-devtools/package-file"
+import {
+  ProjectPath,
+  getProjectPath
+} from "~features/extension-devtools/project-path"
 import {
   TemplatePath,
   getTemplatePath
 } from "~features/extension-devtools/template-path"
 import { definedTraverse } from "~features/helpers/traverse"
-import { Scaffolder } from "~features/manifest-factory/scaffolder"
 
-import { UILibrary, getUILibrary } from "./ui-library"
+import { Scaffolder } from "./scaffolder"
+import { SupportedUIExt, UILibrary, getUILibrary } from "./ui-library"
 
 export const autoPermissionList: ManifestPermission[] = ["storage"]
 
 export abstract class BaseFactory<
   T extends ExtensionManifest | ExtensionManifestV2 = any
 > {
   envConfig: EnvConfig
-  commonPath: CommonPath
-  templatePath: TemplatePath
+
+  #commonPath: CommonPath
+  public get commonPath(): CommonPath {
```

Comment:
can these be public variables instead?

Reply:
No - we don't want them to be mutated outside of this class.